src/startExtension.ts

```diff
@@ -0,0 +1,145 @@
+import * as vscode from "vscode"
+import { getNonce } from "./getNonce"
+// import * as path from "path"
+// import * as fs from "fs"
+
+export class StartExtension {
+  /**
+   * Track the currently panel. Only allow a single panel to exist at a time.
+   */
+  public static currentPanel: StartExtension | undefined
+
+  public static readonly viewType = "temporal-debugger-plugin"
+
+  private readonly _panel: vscode.WebviewPanel
+  private readonly _extensionUri: vscode.Uri
+  private _disposables: vscode.Disposable[] = []
+
+  public static async createOrShow(extensionUri: vscode.Uri): Promise<any> {
+    const column = vscode.window.activeTextEditor ? vscode.window.activeTextEditor.viewColumn : undefined
+
+    // If we already have a panel, show it.
+    if (StartExtension.currentPanel) {
+      StartExtension.currentPanel._panel.reveal(column)
+      StartExtension.currentPanel._update()
+      return
+    }
+
+    // Otherwise, create a new panel.
+    const panel = vscode.window.createWebviewPanel(
+      StartExtension.viewType,
+      "VSinder",
+      column || vscode.ViewColumn.One,
+      {
+        // Enable javascript in the webview
+        enableScripts: true,
+
+        // And restrict the webview to only loading content from our extension's `media` directory.
+        localResourceRoots: [
+          vscode.Uri.joinPath(extensionUri, "media"),
+          vscode.Uri.joinPath(extensionUri, "out/compiled"),
+        ],
+      },
+    )
+
+    StartExtension.currentPanel = new StartExtension(panel, extensionUri)
+  }
+
+  public static async kill(): Promise<any> {
+    await StartExtension.currentPanel?.dispose()
+    StartExtension.currentPanel = undefined
+  }
+
+  public static revive(panel: vscode.WebviewPanel, extensionUri: vscode.Uri): void {
+    StartExtension.currentPanel = new StartExtension(panel, extensionUri)
+  }
+
+  //@typescript-eslint/no-floating-promises
+  private constructor(panel: vscode.WebviewPanel, extensionUri: vscode.Uri) {
+    this._panel = panel
+    this._extensionUri = extensionUri
+
+    // Set the webview's initial html content
+    this._update()
+
+    // Listen for when the panel is disposed
+    // This happens when the user closes the panel or when the panel is closed programatically
+    this._panel.onDidDispose(() => this.dispose(), null, this._disposables)
+  }
+
+  public async dispose(): Promise<any> {
+    StartExtension.currentPanel = undefined
+
+    // Clean up our resources
+    this._panel.dispose()
+
+    while (this._disposables.length) {
+      const x = this._disposables.pop()
+      if (x) {
+        await x.dispose()
+      }
+    }
+  }
+
+  private _update(): void {
+    const webview = this._panel.webview
+
+    this._panel.webview.html = this._getHtmlForWebview(webview)
+    webview.onDidReceiveMessage(async (data) => {
```

Comment:
This probably shouldn't be in `_update`. If I understand the intent correctly, `_update` might be called multiple times and there's no reason to reregister the event handler when that happens.